core/src/main/scala/scalaz/std/List.scala

```diff
@@ -67,17 +67,11 @@ trait ListInstances extends ListInstances0 {
         }
         (a, b) => loop(a, b, Nil)
       }
-      def traverseImpl[F[_], A, B](l: List[A])(f: A => F[B])(implicit F: Applicative[F]) = {
-        val revOpt: Maybe[F[List[B]]] =
-          F.unfoldrOpt[List[A], B, List[B]](l)(_ match {
-            case a :: as => Maybe.just((f(a), as))
-            case Nil => Maybe.empty
-          })(Reducer.ReverseListReducer[B])
 
-        val rev: F[List[B]] = revOpt getOrElse F.point(Nil)
-
-        F.map(rev)(_.reverse)
-      }
+      def traverseImpl[F[_], A, B](l: List[A])(f: A => F[B])(implicit F: Applicative[F]) =
+        F.map(l.reverse.foldRight(F.point(l.genericBuilder[B])) { (a, flb) =>
```

Comment:
In a single scientific test this builder based traverse ran 15% faster than the original.